packages/scripts/src/upgrade-tests/register-new-contract/2_Verify.ts

```diff
@@ -0,0 +1,173 @@
+// the relay transactions have been sent. It can be run on a local hardhat node fork of the mainnet or can be run
+// directly on the mainnet to verify. To run this on the localhost first fork mainnet into a local hardhat node by running:
+// HARDHAT_CHAIN_ID=1 yarn hardhat node --fork https://mainnet.infura.io/v3/<YOUR-INFURA-KEY> --port 9545 --no-deploy
+// Then execute the script from core with the PROPOSAL_DATA logged by  ./src/upgrade-tests/register-new-contract/1_Propose.ts:
+// PROPOSAL_DATA=<PROPOSAL_DATA> yarn hardhat run ./src/upgrade-tests/register-new-contract/2_Verify.ts --network localhost
+
+const hre = require("hardhat");
+const assert = require("assert").strict;
+
+const { TransactionDataDecoder } = require("@uma/financial-templates-lib");
+
+const { RegistryRolesEnum, interfaceName } = require("@uma/common");
+const { getAddress } = require("@uma/contracts-node");
+
+import { Finder, Governor, GovernorHub, GovernorRootTunnel, Registry } from "@uma/contracts-node/typechain/core/ethers";
+import { getContractInstance } from "../../utils/contracts";
+
+// CONSTANTS
+const newContractName = interfaceName.OptimisticAsserter;
+
+function decodeData(data: string) {
+  return TransactionDataDecoder.getInstance().decodeTransaction(data);
+}
+
+const verifyGovernanceHubMessage = async (
+  governorHub: GovernorHub,
+  relayProposal: RelayTransaction,
+  fromBlock: number
+) => {
+  const relayedTransactions = await governorHub.filters.RelayedGovernanceRequest(
+    Number(relayProposal.transaction.params?.chainId),
+    undefined,
+    undefined,
+    undefined
+  );
+
+  const events = await governorHub.queryFilter(relayedTransactions, fromBlock, "latest");
+
+  const found = events.find(
+    (e) =>
+      e.args.calls[0].data === relayProposal.transaction.params?.calls[0].data &&
+      e.args.calls[0].to == relayProposal.transaction.params?.calls[0].to
+  );
+  assert(found, "Could not find RelayedGovernanceRequest matching expected relayed message");
+};
+
+const verifyGovernanceRootTunnelMessage = async (
+  governorRootTunnel: GovernorRootTunnel,
+  relayProposal: RelayTransaction,
+  fromBlock: number
+) => {
+  const relayedTransactions = await governorRootTunnel.filters.RelayedGovernanceRequest(
+    relayProposal.transaction.params.to,
+    undefined
+  );
+  const events = await governorRootTunnel.queryFilter(relayedTransactions, fromBlock, "latest");
+
+  assert(
+    events.find((e) => e.args.data === relayProposal.transaction.params?.data),
+    "Could not find RelayedGovernanceRequest matching expected relayed transaction"
+  );
+};
+
+interface ProposedTransaction {
+  to: string;
+  data: string;
+  value: string;
+}
+
+interface RelayTransaction {
+  to: string;
+  transaction: {
+    name: string;
+    params: {
+      to: string;
+      data?: string;
+      chainId?: string;
+      calls: { to: string; data: string }[];
+    };
+  };
+}
```

Comment:
are these structured repeated elsewhere and can be re-used? they look like common cross-chain send transaction structures.

Reply:
I've moved them to the common file as this types are custom